service/quota.py

```diff
@@ -1,16 +1,106 @@
 from threepio import logger
 
-from core.models import IdentityMembership, Identity, Provider
-
+from core.models import IdentityMembership, Identity
+from core.models.quota import (
+    has_floating_ip_count_quota,
+    has_port_count_quota,
+    has_instance_count_quota,
+    has_cpu_quota,
+    has_mem_quota,
+    has_storage_quota,
+    has_storage_count_quota,
+    has_snapshot_count_quota
+    )
 from service.cache import get_cached_driver
 from service.driver import get_account_driver
 
+
+def check_over_instance_quota(
+        username, identity_uuid,
+        esh_size=None, action=None):
+    """
+    Checks quota based on current limits (and an instance of size, if passed).
+
+    return or raise exc
+    """
+    membership = IdentityMembership.objects.get(
+        identity__uuid=identity_uuid,
+        member__name=username)
+    quota = membership.quota
+    identity = membership.identity
+    driver = get_cached_driver(identity=identity)
```

Comment:
Would we benefit from passing `membership` into this method? It _could_ make it easier to test in the future (since it would remove the method's dependence on the data. (I don't know the usage/call-site, so that might make it not worth the abstraction)